operator/pkg/controller/karmada/planner.go

```diff
@@ -83,10 +84,44 @@ func recognizeActionFor(karmada *operatorv1alpha1.Karmada) Action {
 func (p *Planner) Execute() error {
 	klog.InfoS("Start execute the workflow", "workflow", p.action, "karmada", klog.KObj(p.karmada))
 
+	if err := p.preRunJob(); err != nil {
+		return err
+	}
 	if err := p.job.Run(); err != nil {
+		if err := p.runJobErr(err); err != nil {
+			return err
+		}
+		return err
+	}
+	if err := p.afterRunJob(); err != nil {
 		return err
 	}
 
 	klog.InfoS("Successfully executed the workflow", "workflow", p.action, "karmada", klog.KObj(p.karmada))
 	return nil
 }
+
+func (p *Planner) preRunJob() error {
+	if p.action == InitAction {
+		operatorv1alpha1.KarmadaInProgressing(p.karmada, operatorv1alpha1.Ready, "karmada init job is in progressing")
+	}
+	if p.action == DeInitAction {
+		operatorv1alpha1.KarmadaInProgressing(p.karmada, operatorv1alpha1.Ready, "karmada deinit job is in progressing")
+	}
+
+	return p.Client.Status().Update(context.TODO(), p.karmada)
+}
+
+func (p *Planner) runJobErr(err error) error {
+	operatorv1alpha1.KarmadaFailed(p.karmada, operatorv1alpha1.Ready, err.Error())
+	return p.Client.Status().Update(context.TODO(), p.karmada)
+}
+
+func (p *Planner) afterRunJob() error {
+	operatorv1alpha1.KarmadaCompleted(p.karmada, operatorv1alpha1.Ready, "karmada init job was completed")
+
+	// the cr will also be deleted with karmada is be deleted, so we need not to
```

Comment:
So do you need to update the karmada status?

Reply:
I think we need not to update karmada status, Because karmada cr will be deleted immediately.

Reply:
```
return p.Client.Status().Update(context.TODO(), p.karmada)
```
So the above code is unnecessary？

Reply:
It's necessary, this is init workflow. 
It's unnecessary in `deinit` workflow, I update the comment.

Reply:
updated.